# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1550702518 0
#      Wed Feb 20 22:41:58 2019 +0000
# Node ID f3523eb557911c347de43a05a2b1593aaef1ec4d
# Parent  a89aa5b95a59a8f3f0bcd1e3a32fa9f90752f551
fix name of parent package in nested cdef_package objects

* ov-classdef.cc (cdef_manager::find_package): Use rfind instead of
find to locate the last separator in a nested package name like A.B.C
so that the parent name for package C is A.B, not A.

diff --git a/libinterp/octave-value/ov-classdef.cc b/libinterp/octave-value/ov-classdef.cc
--- a/libinterp/octave-value/ov-classdef.cc
+++ b/libinterp/octave-value/ov-classdef.cc
@@ -3933,17 +3933,17 @@ cdef_manager::find_package (const std::s
         error ("invalid package `%s'", name.c_str ());
     }
   else
     {
       octave::load_path& lp = m_interpreter.get_load_path ();
 
       if (load_if_not_found && lp.find_package (name))
         {
-          size_t pos = name.find ('.');
+          size_t pos = name.rfind ('.');
 
           if (pos == std::string::npos)
             retval = make_package (name, "");
           else
             {
               std::string parent_name = name.substr (0, pos);
 
               retval = make_package (name, parent_name);
